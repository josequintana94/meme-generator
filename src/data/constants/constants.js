export const sizeItems = [   
  {value: 24, text: "24px"},
  {value: 28, text: "28px"},
  {value: 32, text: "32px"},
  {value: 36, text: "36px"},
  {value: 40, text: "40px"},
  {value: 42, text: "42px"},
  {value: 44, text: "44px"},
  {value: 48, text: "48px"},
  {value: 50, text: "50px"}, 
];

export const fontItems = [   
  {value: 1, text: "Georgia"},
  {value: 2, text: "Times New Roman"},
  {value: 3, text: "Arial"},
  {value: 4, text: "Helvetica"},   
  {value: 5, text: "Arial Black"},   
  {value: 6, text: "Impact"},   
  {value: 7, text: "Lucida Sans Unicode"},   
  {value: 8, text: "Tahoma"},   
  {value: 9, text: "Verdana"},   
  {value: 10, text: "Courier New"},   
  {value: 11, text: "Lucida Console"},   
  {value: 12, text: "serif"},   
  {value: 13, text: "sans-serif"},  
  {value: 14, text: "cursive"}, 
  {value: 15, text: "fantasy"}, 
  {value: 16, text: "monospace"},      
];