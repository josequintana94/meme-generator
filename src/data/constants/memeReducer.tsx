import {MemeAction} from "../enum";
import {IinitialStateProps} from "../interface";

export const initialState:IinitialStateProps = {
  topText: "TOP IMAGE TEXT",
  bottomText: "BOTTOM IMAGE TEXT",    
  previewImage: "", 
  backgroundImage:"/10guylogo.jpg",
  fontSize: "50",
  fontColor:"#ffffff",
  fontName:"6", 
};

export const memeReducer = (state = initialState, action:any) => {

  switch (action.type) { 

    case MemeAction.SetTopText: 
      return {
        ...state,
        topText: action.payload
      }; 

    case MemeAction.SetBottomText: 
      return {
        ...state,
        bottomText: action.payload
      }; 

    case MemeAction.SetBackgroundImage: 
      return {
        ...state,
        backgroundImage: action.payload
      }; 

    case MemeAction.SetPreviewImage: 
      return {
        ...state,
        previewImage: action.payload
      }; 

    case MemeAction.SetFontSize: 
      return {
        ...state,
        fontSize: action.payload
      }; 

    case MemeAction.SetFontColor: 
      return {
        ...state,
        fontColor: action.payload
      }; 
    
    case MemeAction.SetFont: 
      return {
        ...state,
        fontName: action.payload
      }; 
          
    default: return state;
  }
};


















