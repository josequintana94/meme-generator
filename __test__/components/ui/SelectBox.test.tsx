import renderer from "react-test-renderer";
import SelectBox from "../../../src/components/ui/SelectBox";
import {sizeItems} from "../../../src/data/constants/constants";

describe(">>>> SelectBox", () => {
  test("++++ Snapshot SelectBox", () => {
    const component = renderer.create(
      <SelectBox
        id="selectBox"
        items={sizeItems}
        selectedValue=""
        onChange={jest.fn()}                
      />             
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

export {};