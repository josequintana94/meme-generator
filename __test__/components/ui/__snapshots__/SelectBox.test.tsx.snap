// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>> SelectBox ++++ Snapshot SelectBox 1`] = `
<select
  id="selectBox"
  onChange={[MockFunction]}
  value=""
>
  <option
    value={24}
  >
    24px
  </option>
  <option
    value={28}
  >
    28px
  </option>
  <option
    value={32}
  >
    32px
  </option>
  <option
    value={36}
  >
    36px
  </option>
  <option
    value={40}
  >
    40px
  </option>
  <option
    value={42}
  >
    42px
  </option>
  <option
    value={44}
  >
    44px
  </option>
  <option
    value={48}
  >
    48px
  </option>
  <option
    value={50}
  >
    50px
  </option>
</select>
`;
